ql/src/java/org/apache/hadoop/hive/ql/ddl/database/drop/DropDatabaseAnalyzer.java

```diff
@@ -61,7 +62,7 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     boolean isDbLevelLock = true;
     if (cascade) {
       try {
-        List<Table> tables = db.getAllTableObjects(databaseName);
+        List<Table> tables = db.getAllTableObjects(databaseName, true);
```

Comment:
There is a `hive.metastore.client.filter.enabled` to disable the client filter, we can use it instead of depend on a new method, e.g
```java
try {
   HiveConf hiveConf = new HiveConf(conf);
   conf.set("hive.metastore.client.filter.enabled", "false");
   Hive db = Hive.get(hiveConf);
   ....
} finally {
   // restore
   db = Hive.get(conf);
 }

Reply:
I'll implement this suggestion as setting config in session is not taken into effect. we don't need a finally block since we are creating a new variable 'db' instead of overriding the already existing one. Also, the existing variable 'db' is a final variable so we cannot override it.

Reply:
The new created `db` will overwrite the thread-local hive, and the underlying client disables the client filtering, if we use this client for other calls in the future, it may cause problems.